Flaky test for dashboard provisioning

<!--
Please use this template to create your bug report. By providing as much info as possible you help us understand the issue, reproduce it and resolve it for you quicker. Therefore take a couple of extra minutes to make sure you have provided all info needed.

PROTIP: record your screen and attach it as a gif to showcase the issue.

- Questions should be posted to: https://community.grafana.com
- Use query inspector to troubleshoot issues: https://bit.ly/2XNF6YS
- How to record and attach gif: https://bit.ly/2Mi8T6K
-->

**What happened**:

```
--- FAIL: TestUpdatingProvisionionedDashboards (4.56s)
testinfra.go:31: Grafana is listening on 127.0.0.1:38641
--- FAIL: TestUpdatingProvisionionedDashboards/when_provisioned_directory_is_not_empty,_dashboard_should_be_created (0.00s)
	        api_dashboards_test.go:149: 
	            	Error Trace:	api_dashboards_test.go:149
            	Error:      	Received unexpected error:
          	            	Get "http://admin:***@127.0.0.1:38641/api/search?query=Grafana+Dev+Overview+%26+Home": dial tcp 127.0.0.1:38641: connect: connection refused

Test:       	TestUpdatingProvisionionedDashboards/when_provisioned_directory_is_not_empty,_dashboard_should_be_created
testinfra.go:53: Server exited uncleanly error *provisioning.ProvisioningServiceImpl run error: Failed to create provisioner: Failed to read dashboards config: failed to provision dashboards with "provisioned dashboards" reader: failed to check whether org. with the given ID exists: database table is locked
```

from https://drone.grafana.net/grafana/grafana-enterprise/33539/2/14

**What you expected to happen**:

✅

**How to reproduce it (as minimally and precisely as possible)**:

Haven't tried making a reliable reproduction step yet, it's due to a flaky test.

**Anything else we need to know?**:

**Environment**:
- Grafana version: main
- Data source type & version: n/a
- OS Grafana is installed on: n/a
- User OS & Browser: n/a
- Grafana plugins: n/a
- Others: CI flakiness

--- a/pkg/tests/testinfra/testinfra.go
+++ b/pkg/tests/testinfra/testinfra.go
@@ -314,12 +314,15 @@ func CreateGrafDir(t *testing.T, opts ...GrafanaOpts) (string, string) {
 			_, err = logSection.NewKey("address", o.GRPCServerAddress)
 			require.NoError(t, err)
 		}
+		// retry queries 3 times by default
+		queryRetries := 3
 		if o.QueryRetries != 0 {
-			logSection, err := getOrCreateSection("database")
-			require.NoError(t, err)
-			_, err = logSection.NewKey("query_retries", fmt.Sprintf("%d", o.QueryRetries))
-			require.NoError(t, err)
+			queryRetries = int(o.QueryRetries)
 		}
+		logSection, err := getOrCreateSection("database")
+		require.NoError(t, err)
+		_, err = logSection.NewKey("query_retries", fmt.Sprintf("%d", queryRetries))
+		require.NoError(t, err)
 	}
 
 	cfgPath := filepath.Join(cfgDir, "test.ini")

